paasta_tools/mesos_maintenance.py

```diff
@@ -599,7 +602,10 @@ def undrain(hostnames):
     :returns: None
     """
     log.info("Undraining: %s" % hostnames)
-    unreserve_all_resources(hostnames)
+    try:
+        unreserve_all_resources(hostnames)
```

Comment:
This is the one potentially risky part. If we fail to unreserve the resources, but mark the host as being up, we will be paying for resources that can't be used. It probably makes sense to also update the maintenance cleanup scripts to check for dynamic resource reservations made for the purpose of maintenance that exist on boxes marked as being 'up' and then remove them.

Reply:
I think it already does right? i.e. when it gets to: https://github.com/Yelp/paasta/blob/master/paasta_tools/cleanup_maintenance.py#L103
the cleanup script should spot anything where we managed to undrain it but where we haven't unreserved the resources.